File: store/android/src/main/java/com/jakewharton/sdksearch/store/DbModule.java
package com.jakewharton.sdksearch.store;

import android.arch.persistence.db.SupportSQLiteOpenHelper;
import android.arch.persistence.db.SupportSQLiteOpenHelper.Configuration;
import android.arch.persistence.db.framework.FrameworkSQLiteOpenHelperFactory;
import android.content.Context;
import android.support.annotation.Nullable;
import com.squareup.sqlbrite3.BriteDatabase;
import com.squareup.sqlbrite3.SqlBrite;
import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import io.reactivex.Scheduler;
import timber.log.Timber;

@Module //
abstract class DbModule {
  @Provides static BriteDatabase briteDatabase(Context context, @Nullable String filename, Scheduler scheduler) {
    Configuration configuration = Configuration.builder(context)
        .callback(new DbCallback())
        .name(filename)
        .build();
    SupportSQLiteOpenHelper helper = new FrameworkSQLiteOpenHelperFactory().create(configuration);
    SqlBrite.Logger loger = message -> Timber.tag("DB").d(message);
    SqlBrite sqlBrite = new SqlBrite.Builder().logger(loger).build();
    return sqlBrite.wrapDatabaseHelper(helper, scheduler);
  }

  @Binds abstract ItemStore itemStore(SqlItemStore impl);
}

File: store/android/src/main/java/com/jakewharton/sdksearch/store/DbModule.kt
package com.jakewharton.sdksearch.store

import android.arch.persistence.db.SupportSQLiteOpenHelper.Configuration
import android.arch.persistence.db.framework.FrameworkSQLiteOpenHelperFactory
import android.content.Context
import com.squareup.sqlbrite3.BriteDatabase
import com.squareup.sqlbrite3.SqlBrite
import dagger.Module
import dagger.Provides
import io.reactivex.Scheduler
import timber.log.Timber

@Module //
internal class DbModule {
  @Provides fun briteDatabase(context: Context, filename: String?, scheduler: Scheduler): BriteDatabase {
    val configuration = Configuration.builder(context)
        .callback(DbCallback())
        .name(filename)
        .build()
    val helper = FrameworkSQLiteOpenHelperFactory().create(configuration)
    val logger = SqlBrite.Logger { message -> Timber.tag("DB").d(message) }
    val sqlBrite = SqlBrite.Builder().logger(logger).build()
    return sqlBrite.wrapDatabaseHelper(helper, scheduler)
  }

  @Provides fun itemStore(db: BriteDatabase): ItemStore = SqlItemStore(db)
}

